src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.cpp

```diff
@@ -2117,7 +2126,60 @@ namespace UiaOperationAbstraction
         return AsType<UiaString>();
     }
 
-    UiaVariant::operator std::shared_ptr<wil::unique_variant>() const
+    UiaBool UiaVariant::IsElement() const

```

Comment:
So this is basically `IsType`, except in the local case we do a `QueryInterface` for IUIAutomationElement? Rather than rewriting `IsType`, here are my thoughts:

We could rewrite `IsElement` as follows:

```cpp
const bool isElement = IsType<UiaElement>();
if (!ShouldUseRemoteApi() && isElement)
{
    // Make sure the IUnknown is actually an IUIAutomationElement.
    auto localValue = std::get<typename LocalType>(m_member);
    winrt::com_ptr<IUnknown> unknown;
    unknown.copy_from(localValue->punkVal);
    return !!unknown.try_as<IUIAutomationElement>();
}
else
{
    return false;
}
```

Or, we could add a `c_isLocalType` function to each type to do the work for us.

Personally I think I prefer the first option in this case. What do you think?